catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavEntry.java

```diff
@@ -299,6 +299,16 @@ public void setDirectory(final boolean directory) {
     this.directory = directory;
   }
 
+  /**
+   * If this {@code DavEntry}'s file is cached, delete it. A {@code DavEntry}'s file is cached on
+   * the first call to {@link #getFile(Sardine) DavEntry.getFile(Sardine)}.
+   */
+  public void deleteCacheIfExists() {
```

Comment:
Does anything else use this method? Not sure why we need the null check here (or method) if not since the above if already checks for a null file

Reply:
I put this check because the `file` is null until after the first call to `getFile(DavEntry)`. So if someone called this method before caching `file.getParentFile()` would produce an NPE.